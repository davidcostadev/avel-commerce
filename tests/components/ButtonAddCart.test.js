/* eslint no-undef: "off" */
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import moxios from 'moxios'
import ButtonAddCartConnected, { ButtonAddCart } from '../../components/product/ButtonAddCart'
import { cartProductIns, cartProductInsError } from '../api/__mocks__/Cart.test.mocks'

Enzyme.configure({ adapter: new Adapter() })

describe('should', () => {
  const initialState = {}
  const mockStore = configureStore()
  let store

  afterEach(() => moxios.uninstall())

  beforeEach(() => {
    store = mockStore(initialState)
    moxios.install()
  })

  it('Add product on cart', () => {
    const tree = renderer.create((
      <ButtonAddCartConnected
        productId={1}
        store={store}
      />
    )).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('click on button', (done) => {
    const onSetCart = jest.fn()
    const onSetCartItens = jest.fn()

    process.env = {
      PASSKEY: '123',
      DOMAIN_API: 'http://domain.com',
    }

    const wrapper = shallow((
      <ButtonAddCart
        productId={1}
        onSetCart={onSetCart}
        onSetCartItens={onSetCartItens}
      />
    ))

    moxios.stubRequest('http://domain.com/TSvmCarrinho/sp_web_carrinho_produto_ins', {
      status: 200,
      responseText: {
        result: [cartProductIns],
      },
    })

    wrapper.find('button').simulate('click')

    moxios.wait(() => {
      expect(onSetCart).toMatchSnapshot()
      expect(onSetCartItens).toMatchSnapshot()
      done()
    })
  })

  it('fall on click on button', (done) => {
    const onSetCart = jest.fn()
    const onSetCartItens = jest.fn()
    console.error = jest.fn()

    process.env = {
      PASSKEY: '123',
      DOMAIN_API: 'http://domain.com',
    }
    const wrapper = shallow((
      <ButtonAddCart
        productId={1}
        onSetCart={onSetCart}
        onSetCartItens={onSetCartItens}
      />
    ))

    moxios.stubRequest('http://domain.com/TSvmCarrinho/sp_web_carrinho_produto_ins', {
      status: 200,
      responseText: {
        result: [cartProductInsError],
      },
    })

    wrapper.find('button').simulate('click')

    moxios.wait(() => {
      expect(console.error).toMatchSnapshot()
      done()
    })
  })
})
