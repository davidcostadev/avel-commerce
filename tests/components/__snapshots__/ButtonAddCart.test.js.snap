// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Add product on cart 1`] = `
<button
  className="btn btn-lg btn-primary btn-buy"
  onClick={[Function]}
  type="button"
>
  Comprar
</button>
`;

exports[`should click on button 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "PS_ANOTACOES": "",
        "PS_CEP": "0",
        "PS_CODIGOP": "",
        "PS_DATADOCADASTRO": "09/02/2018 21:46:32",
        "PS_FEEDBACK": "As informações carregadas com sucesso",
        "PS_ID_CARRINHO": "12218",
        "PS_ID_CLIENTE": "0",
        "PS_ID_ERRO": "0",
        "PS_ID_PG_FORMA": "13",
        "PS_ID_TIPO_FRETE": "41068",
        "PS_ID_VENDEDOR": "",
        "PS_PG_FORMA": "PAGSEGURO/BOLETO",
        "PS_PRAZO_ENTREGA_DIA": "",
        "PS_QUANTIDADESDEITENS": "1",
        "PS_SESSAO": "c2fcde6b184dfc25ff28f8a5cf4b1da8",
        "PS_TIPO_FRETE": "CORREIO PAC",
        "PS_TX_DESCONTO_CODP": "",
        "PS_VL_DELIVERY": "10",
        "PS_VL_DESCONTO": "0",
        "PS_VL_FRETE": "0",
        "PS_VL_PARCELA": "",
        "PS_VL_SUBTOTAL": "21",
        "PS_VL_TOTAL_GERAL": "21",
        "PS_VL_TOTAL_PARCELADO": "",
      },
    ],
  ],
}
`;

exports[`should click on button 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "PS_ALTURA_MM": "0",
          "PS_COMPRIMENTO_MM": "0",
          "PS_DATADOCADASTRO": "09/02/2018 21:46:32",
          "PS_DESCRICAO": "Graxa de Silicone 50g IMPLASTEC",
          "PS_DIAMETRO_MM": "0",
          "PS_ESTOQUE_DEPOSITO": "0",
          "PS_ESTOQUE_FILIA": "10",
          "PS_ESTOQUE_LOJA": "11",
          "PS_ESTOQUE_TOTAL": "21",
          "PS_FORMATO_ENVIO": "Indefindo",
          "PS_ID_CARRINHO": "12218",
          "PS_ID_IMAGEM": "168058",
          "PS_ID_MOVIMENTO_CAR": "22426",
          "PS_ID_PRODUTO": "25775",
          "PS_IMAGEM_TXT_ALT": "Graxa de Silicone 50g IMPLASTEC",
          "PS_IMAGEM_TXT_TITLE": "Graxa de Silicone 50g IMPLASTEC",
          "PS_LARGURA_MM": "0",
          "PS_LINK_TITLE": "Acesse maiores informações  Graxa de Silicone 50g IMPLASTEC",
          "PS_PATH_IMAGEM_60": "http://dominio.com/a.jpg",
          "PS_PATH_IMAGEM_ZOMM": "http://dominio.com/b.jpg",
          "PS_PATH_PAGE": "",
          "PS_PESO": "0",
          "PS_QT": "1",
          "PS_REGISTROS_COUNT": "1",
          "PS_VALOR_DE_VENDA": "21",
          "PS_VALOR_TOTAL": "21",
          "PS_VL_UNITARIO": "21",
        },
      ],
    ],
  ],
}
`;

exports[`should fall on click on button 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Error: Não foi possível carregar as informações do produto],
    ],
  ],
}
`;
